// ngx-admin-lte/src/widgets/menu-aside/menu-aside.component.ts
import { Component, OnInit, Input } from '@angular/core';
import { User } from '../../models/user';
import { Router } from '@angular/router';
import { UserService } from '../../services/user.service';
declare var $: any;

@Component({
  selector: 'app-menu-aside',
  styleUrls: ['./menu-aside.component.css'],
  templateUrl: './menu-aside.component.html'
})
export class MenuAsideComponent implements OnInit {
  public currentUrl: string;
  public currentUser: User = new User();

  @Input() public links: Array<any> = [];

  constructor(private userServ: UserService, public router: Router) {
    // getting the current url
    this.router.events.subscribe((evt: any) => this.currentUrl = evt.url);
    this.userServ.currentUser.subscribe((user) => this.currentUser = user);
  }

  public ngOnInit() {
    // TODO
      $(document).on("click", ".treeview>ul>li>a", function(){
           console.log("heightSet");
             // close menu on load
            $("body").removeClass("sidebar-open");
      });
  }

}
