// ngx-admin-lte/src/services/can-activate-guard.service.ts
import { Injectable } from '@angular/core';
import { CanActivate, Router } from '@angular/router';
import { UserService } from './user.service';
import { User } from '../models/user';

@Injectable()
export class CanActivateGuard implements CanActivate {
  private connected = false;

  constructor(
    private router: Router,
    private userService: UserService
  ) {
    this.userService.currentUser.subscribe((user) => {
      this.connected = user.connected;
    });
  }

  public canActivate() {
     
    // test here if you user is logged
    if ( !this.connected ) {
     if (localStorage.getItem('ssdUser')) {
         
            let userData = JSON.parse(localStorage.getItem('ssdUser'));
            let lastNameUser = userData.lastname;
            let user1 = new User( {
                      avatarUrl: 'public/assets/img/user2-160x160.jpg',
                      email: userData.username,
                      firstname: userData.firstname,
                      lastname: lastNameUser
                  });

          user1.connected = true;

          this.userService.setCurrentUser( user1 );
            // logged in so return true
            return true;
        }
        
        
      this.router.navigate( [ 'login' ] );
    }
    return this.connected;
  }
}
